### DISCUSSION TITLE
What is the reasoning behind the name of the "done" status?

### DISCUSSION THREAD

**Discussion by axelsimon** [GothenburgBitFactory/taskwarrior#2412] [Q&A]
Hi,

Here's a thing that creates a bit of friction and unnecessary cognitive load:
- to complete a task, you mark it as "done"
- this puts in the status "completed"
- and gives an "ended" date

Why do we have 3 different names for what is mostly the same concept: "this is no longer needs to be done"?

It would appear that this could be reduced somewhat to express the concept in a way that would be used consistently throughout taskwarrior.

I can see how ended is different from done, a deleted task is ended without having be done. Fair enough. But "completed" and "done" are really very similar, and look like they could be folded into one term (likely the shorter one, "done"). Or am i missing something?

I think this would particularly help new users trying to wrap their head around the different verbs/words that need to be remembered to use taskwarrior.

**Comment by tbabej** [comment]
The different names are there to not overload a single word with different meanings:
* _done_ is a command that makes modification to a task (namely, changes its status to _completed_)
* _completed_ is a task status and a report name
* _end_ is an attribute of a task
